### PR TITLE
Use relative paths with CLI

### LINKED ISSUE #2964: [question] .prettierignore syntax?
I'm trying to use `.prettierignore` file & I'm not sure why it's not working properly? I'm using `1.7.4`

`.prettierignore`

```
path/to/file.*.js
path/to/another/file.js
path/to/folder/
path/**/to/folder/
*.min.*
public/**/
```

```
$ prettier --find-config-path ./
prettier.config.js
```

`prettier.config.js`

```js
module.exports = {
  parser: 'flow',
  singleQuote: true,
  printWidth: 120,
  trailingComma: 'es5',
};
```

if I do this from the command line 

```sh
$ ./node_modules/.bin/prettier --write path/to/folder/*.js
# all .js files gets formatted although the path is in .prettierignore
```

I tired `--config-precedence file-override|prefer-file` but still these files which are inside paths that are ignored are getting formatted. Not sure what is the problem?

### PR DESCRIPTION
**Pull request by ahmedelgabri**
Fixes #2964

@ikatyang I need some pointers for the tests. I thought about adding another test inside [`tests_integration/__tests__/ignore-absolute-path.js`](https://github.com/prettier/prettier/blob/master/tests_integration/__tests__/ignore-absolute-path.js)

### CODE REVIEW

**Comment by ikatyang** [comment]
I think you can add them in a new file `tests_integration/__tests__/ignore-relative-path.js` and add their fixtures in `tests_integration/cli/ignore-relative-path`.

File: tests_integration/__tests__/ignore-relative-path.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+"use strict";
+
+const path = require("path");
+const runPrettier = require("../runPrettier");
+
+describe("support relative paths", () => {
+  runPrettier("cli/ignore-relative-path", [
+    path.resolve(__dirname, "../cli/ignore-relative-path/level1/level2/level3/file.js"),
+    path.resolve(__dirname, "../cli/ignore-relative-path/anotherfile.js"),
```
Comment (ikatyang) [inline tests_integration/__tests__/ignore-relative-path.js]: These should be relative paths, `path.resolve()` will make them absolute.

File: tests_integration/cli/ignore-relative-path/anotherfile.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+var x = 'this should be formatterd';
+
```
Comment (ikatyang) [inline tests_integration/cli/ignore-relative-path/anotherfile.js]: Name this file `shouleBeFormat.js` would be better.

File: tests_integration/cli/ignore-relative-path/level1/level2/level3/file.js [hunk 3]
```diff
@@ -0,0 +1 @@
+var x = 'this should not be formatterd';
```
Comment (ikatyang) [inline tests_integration/cli/ignore-relative-path/level1/level2/level3/file.js]: Name this file `shouleNotBeFormat.js` would be better.

File: tests_integration/cli/ignore-relative-path/anotherfile.js (hunk 2 above)
Comment (ikatyang) [reply tests_integration/cli/ignore-relative-path/anotherfile.js]: Or `shouldNotBeIgnored`?

File: tests_integration/__tests__/ignore-relative-path.js (hunk 1 above)
Comment (ahmedelgabri) [reply tests_integration/__tests__/ignore-relative-path.js]: Yes I know, I just wanted to push the changes & will clean it up the whole PR in an hour or two 😊

Comment (ikatyang) [reply tests_integration/__tests__/ignore-relative-path.js]: Let me know if you're ready. 👍

**Comment by ahmedelgabri** [comment]
@ikatyang done, updated the tests & rebased. You can check it now

**Comment by lydell** [comment]
@ahmedelgabri You have a minor lint error that causes the tests to fail :)

```
  ✘  no-unused-vars  'path' is assigned a value but never used  

  tests_integration/__tests__/ignore-relative-path.js:3:7

  const path = require("path");

         ^
```

**Comment by ahmedelgabri** [comment]
Oops, @lydell should be fixed now :)

**Comment by azz** [comment]
Thanks!

**Pull request merged**